### PR TITLE
Fixed axe violation for keyboard accessibility of scrollable content

### LINKED ISSUE #941: Axe Violation for keyboard accessibility of scrollable content.
## Description
Axe Core fails in certain components giving violation for keyboard accessibility.

## Error Report
```
Elements that have scrollable content should be accessible by keyboard 
```
## Components

- Dialog   Modal
- Infinite List
- Menu
- Collapsible Menu
- Popup

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fixes #941

### CODE REVIEW

File: packages/terra-dialog-modal/src/DialogModal.module.scss [hunk 1]
```diff
@@ -71,15 +71,23 @@
     width: 100%; // Needed for IE to wrap content correctly in flexbox
   }
 
+  .dialog-modal-container:focus {
```
Comment (dkasper-was-taken) [inline packages/terra-dialog-modal/src/DialogModal.module.scss]: Disabling the focus outline needs to be evaluated by UX/Accessibility to determine if focus is needed, and UX needs to determine if we want a stylized focus variant as well to match our themes.

File: packages/terra-dialog-modal/src/DialogModal.module.scss [hunk 2]
```diff
@@ -71,15 +71,23 @@
...
-
+  
   .dialog-modal-body {
     background-color: var(--terra-dialog-modal-body-background-color);
     flex: 0 1 auto;
     overflow-y: auto;
     padding: var(--terra-dialog-modal-body-padding, 0.7142857143rem);
   }
+
+  .dialog-modal-body:focus {
```
Comment (dkasper-was-taken) [inline packages/terra-dialog-modal/src/DialogModal.module.scss]: Same as above.

File: packages/terra-dialog-modal/src/DialogModal.module.scss (hunk 1 above)
Comment (saket2403) [reply packages/terra-dialog-modal/src/DialogModal.module.scss]: cc @neilpfeiffer

Comment (neilpfeiffer) [reply packages/terra-dialog-modal/src/DialogModal.module.scss]: @dkasper-was-taken - in terms of transferring programatic focus to the non-interactive parent content areas of our disclosure containers, adding `outline: none;` (no theme-ability needed) has been our standard practice like on [popup](https://github.com/cerner/terra-framework/blob/6cf7f8722f57836915c06589297b30efd09fb1df/packages/terra-popup/src/PopupContent.module.scss#L8), [slide-panel](https://github.com/cerner/terra-framework/blob/6cf7f8722f57836915c06589297b30efd09fb1df/packages/terra-slide-panel/src/SlidePanel.module.scss#L12), [abstract-modal](https://github.com/cerner/terra-framework/blob/6cf7f8722f57836915c06589297b30efd09fb1df/packages/terra-abstract-modal/src/AbstractModal.module.scss#L7), [the rollup-menu in tabs](https://github.com/cerner/terra-framework/blob/6cf7f8722f57836915c06589297b30efd09fb1df/packages/terra-tabs/src/StructuralTabs.module.scss#L186), etc.

This looks correct @saket2403

**Comment by neilpfeiffer** [comment]
+1 UX Review 

@dkasper-was-taken @tbiethman @mjhenkes @ryanthemanuel this is ready for maintainer review.

**Comment by benbcai** [comment]
These [divs](https://github.com/cerner/terra-framework/blob/master/packages/terra-slide-panel/src/SlidePanel.jsx#L113-L119) wrapping the main content and panel content in terra-slide-panel explicitly set the tabIndex to -1 and is also causing this violation. Should the tabIndex be set to 0 and hide the focus glow?

cc @neilpfeiffer @saket2403

**Comment by saket2403** [comment]
> These [divs](https://github.com/cerner/terra-framework/blob/master/packages/terra-slide-panel/src/SlidePanel.jsx#L113-L119) wrapping the main content and panel content in terra-slide-panel explicitly set the tabIndex to -1 and is also causing this violation. Should the tabIndex be set to 0 and hide the focus glow?

Slide Panel Updated in ff79bb3 & 069b136

File: packages/terra-dialog-modal/tests/wdio/dialog-modal-spec.js [hunk 3]
```diff
@@ -69,6 +69,8 @@ describe('Dialog Modal', () => {
     Terra.it.matchesScreenshot('focused shifted outside the beginning of the modal', { selector: '#root' });
 
     it('does not focus on interactive content within the app when modal is open and focus is shifted forwards from the modal dialog', () => {
+      browser.refresh();
```
Comment (emilyrohrbough) [inline packages/terra-dialog-modal/tests/wdio/dialog-modal-spec.js]: Why has this refresh been added?

File: packages/terra-infinite-list/src/InfiniteList.jsx [hunk 4]
```diff
@@ -499,6 +499,7 @@ class InfiniteList extends React.Component {
     if (role && role.length > 0 && role !== 'none') {
       attrSpread.role = role;
     }
+    attrSpread.tabIndex = '0';
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/src/InfiniteList.jsx]: Is there a test to validate this change?

File: packages/terra-popup/src/_PopupOverlay.jsx [hunk 5]
```diff
@@ -59,6 +59,8 @@ class PopupOverlay extends React.Component {
           onClick={this.handleOnClick}
           {...customProps}
           className={cx(['popup-overlay', customProps.className])}
+          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+          tabIndex="0"
```
Comment (emilyrohrbough) [inline packages/terra-popup/src/_PopupOverlay.jsx]: Doesn't adding tab index here allow users to tab to the overlay?

**Comment by emilyrohrbough** [comment]
Are there axe tests that need updated or added to validate these changes?

File: packages/terra-popup/src/_PopupOverlay.jsx (hunk 5 above)
Comment (saket2403) [reply packages/terra-popup/src/_PopupOverlay.jsx]: Removed in 4bc7213

File: packages/terra-infinite-list/src/InfiniteList.jsx (hunk 4 above)
Comment (saket2403) [reply packages/terra-infinite-list/src/InfiniteList.jsx]: Added jest in 4bc7213

Comment (saket2403) [reply packages/terra-infinite-list/src/InfiniteList.jsx]: Added jest in 4bc7213

File: packages/terra-dialog-modal/tests/wdio/dialog-modal-spec.js (hunk 3 above)
Comment (saket2403) [reply packages/terra-dialog-modal/tests/wdio/dialog-modal-spec.js]: Removed in db4eea8

**Comment by saket2403** [comment]
> Are there axe tests that need updated or added to validate these changes?

Added Test for validating tabIndex in 4bc7213

File: packages/terra-dialog-modal/src/DialogModal.jsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
```
Comment (mjhenkes) [inline packages/terra-dialog-modal/src/DialogModal.jsx]: If it's absolutely necessary to disable this, disable it for the specific line only.

Comment (saket2403) [reply packages/terra-dialog-modal/src/DialogModal.jsx]: Changed 126317f

**Comment by neilpfeiffer** [comment]
UX REVIEW HOLD: Need to re-review tab focus trapping in slide-panel for various usage situations

**Comment by benbcai** [comment]
Closing this and we will visit how to properly handle accessibility with scrollable content when we tackle #991.

**Pull request closed without merging**